### ISSUE TITLE
tensorflow Load model gives the random prediction in python new session.(LSTM.)

### ISSUE THREAD

**Issue by samrajkodai** [keras-team/keras#17057]
after train and test the model gives the good accuracy and perfect predictions.

model.save()
load_model()
once i save the model and testing in a new python session it gives random prediction.

i tried many methods but I'm not able to solve the issue. anyone have a solution?

i have attached the code also
my tensorflow version is  2.11.0-dev20220812
[issue.txt](https://github.com/tensorflow/tensorflow/files/9597363/issue.txt)

**Comment by samrajkodai** [comment]
@sushreebarsa  

any updates?

**Comment by sushreebarsa** [comment]
@samrajkodai I tried to replicate the issue on colab using TF v2.10, please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/28ecaf1178683852c7768727f7b0d16d/untitled8.ipynb) for reference and let us know if we are missing something as it is showing different error. Thank you!

**Comment by samrajkodai** [comment]
@sushreebarsa 

actually i did'nt attached the csv file my problem is after developing and saving the model it gives good accuracy. if i am test that model in new python session it gives 0% accuracy

# NOTE:

## model before loading : 0.9572490706319703 acc

## model after loading : 0.0030978934324659233 acc

**Comment by gadagashwini** [comment]
Hi @samrajkodai, Could you try the workaround mentioned on similar issue [here](https://github.com/keras-team/keras/issues/4875#issuecomment-1255280893). Thank you!

**Comment by samrajkodai** [comment]
hi @gadagashwini  my problem is not solved i tried this too.

**Comment by samrajkodai** [comment]
hi @gadagashwini any updates?

**Comment by gowthamkpr** [comment]
@samrajkodai Please give us entire code to reproduce this issue. Please follow the steps mentioned [here](https://github.com/keras-team/keras/issues/4875#issuecomment-1255280893) and let me know if it helps. Thanks!!

**Comment by samrajkodai** [comment]
[issue (2).txt](https://github.com/keras-team/keras/files/9712888/issue.2.txt)

hi ,i already tried that too ,it works fine in the same teriminal but when i restart the terminal or open the new jupyter notebook for testing it gives the random prediction only.

**Comment by samrajkodai** [comment]
hi, have you find any solutions?

**Comment by PaulOctopusZLWB** [comment]
Hi, I would like to report the same problem.
I trained a Transformer based model on TF2.5 and it performed well on the training set and validation. 
When I reproduce the model using model.load_weight on a test set in another script, the model does not work at all.
I found that even after predict the results on another script using the validation set, it still performed poorly which can not matching the training matrix. 
In other words, the problem was with the model in the save and load sessions. I have no clue about this.

**Comment by tilakrayal** [comment]
@samrajkodai,
Apologies for the delay. Could you please provide the .csv which you are trying in the code. Also could you please confirm, is the issue happening on only Jupyter or other IDE's as well. Please try on the colab as well with the latest stable v2.13.
And please take a look at these comments from the developer for the similar error/issue.

https://github.com/tensorflow/tensorflow/issues/60192
https://github.com/tensorflow/tensorflow/issues/59281
Thank you!
